examples/plot_alignment_methods_benchmark.py

```diff
@@ -163,11 +145,109 @@
+# Initialize the IdentifiableFastSRM model. This version of SRM ensures that
+# the solution is unique.
+#
+
+from fastsrm.identifiable_srm import IdentifiableFastSRM
+
+srm = IdentifiableFastSRM(
+    n_components=30,
+    n_iter=10,
+)
+
+# #############################################################################
+# For the SRM method, we will use the same training data as before
+# (source subjects: sub-01 and sub-02) and the target subject (sub-04), and we
+# will divide our source and target subjects into training and testing sets,
+# leveraging the AP and PA acquisitions.
+
+# The training set:
+# * source train: AP acquisitions from source subjects (sub-01, sub-02).
+#   These will be projected into a shared response space.
+# * target train: AP acquisitions from the target subject (sub-04).
+#   These define the target shared space.
 #
+
+source_subjects = [sub for sub in sub_ids if sub != "sub-04"]
+source_train = [
+    concat_imgs(df[(df.subject == sub) & (df.acquisition == "ap")].path.values)
+    for sub in source_subjects
+]
+target_train = concat_imgs(
+    df[(df.subject == "sub-04") & (df.acquisition == "ap")].path.values
+)
+
+# The testing set:
+# * source test: acquisitions from source subjects (sub-01, sub-02).
+#   These will be projected into the shared space and transformed to predict
+#   the target.
+# * target test: PA acquisitions from the target subject (sub-04).
+#   These serve as ground truth to evaluate prediction accuracy.
+#
+
+source_test = [
+    concat_imgs(df[(df.subject == sub) & (df.acquisition == "pa")].path.values)
+    for sub in source_subjects
+]
+target_test = concat_imgs(
+    df[(df.subject == "sub-04") & (df.acquisition == "pa")].path.values
+)
+
+################################################################################
+# Fit the SRM model
+# -----------------
+
+# Step 1: Fit SRM on training data from source subjects
+shared_response = srm.fit_transform(
+    [roi_masker.transform(s).T for s in source_train]
+)
+
+# Step 2: Freeze the SRM model and add target subject data. This projects the
+# target subject data into the shared response space.
+srm.aggregate = None
+srm.add_subjects([roi_masker.transform(target_train).T], shared_response)
+
+# Step 3: Use SRM to transform new test data from the target subject
+aligned_test = srm.transform([roi_masker.transform(target_test).T])
+aligned_pred = roi_masker.inverse_transform(
+    srm.inverse_transform(aligned_test[0])[0].T
+)
+
+# Step 4: Evaluate voxelwise correlation between predicted and true test
+# signals.
+srm_error = score_voxelwise(
+    target_test, aligned_pred, masker=roi_masker, loss="corr"
+)
+srm_score = roi_masker.inverse_transform(srm_error)
+
+title = "Correlation of prediction after SRM alignment"
+display = plotting.plot_stat_map(
+    srm_score,
+    display_mode="z",
+    cut_coords=[-15, -5],
+    vmax=1,
+    title=title,
+    axes=axes[len(methods)],
+    colorbar=True,
+)
+
+plt.tight_layout()
+plt.show()
```

Comment:
We won't need this additional plotting, since we'll include it in the first plot as a sub-plot